### PR TITLE
i18n(ja): Add new `docker.mdx` translation

### PR DESCRIPTION
**Pull request by piro-hiroki**
#### Description (required)

Translate `docker.mdx` into Japanese！
file: [recipes/docker.mdx](https://github.com/withastro/docs/blob/main/src/content/docs/en/recipes/docker.mdx)

#### Related issues & labels (optional)

- Closes #
- Suggested label:

### CODE REVIEW

**Comment by piro-hiroki** [comment]
`ja/guides/deploy/google-cloud/` の翻訳が最新の変更に追従してなさそう。。。/ The translation of `en/guides/deploy/google-cloud/` doesn't seem to follow the latest changes....

**Comment by sarah11918** [comment]
Hi @piro-hiroki ! Just checking if this PR should still be marked as a draft?

If it's ready for editing, then please change the status so that it is ready for review! We are hoping to merge as many PRs by next week as we can before the docs site code migrates to Starlight. So if this is ready, we would like to be able to review it.  Thanks!

**Comment by sarah11918** [comment]
Hi @piro-hiroki ! I think I fixed the links on this one! Is it ready for a review?

File: src/content/docs/ja/recipes/docker.mdx [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+Dockerイメージとコンテナは、AWS、Azure、[Google Cloud](/ja/guides/deploy/google-cloud/#cloud-run-ssrと静的配信)のような多くの異なるプラットフォームにデプロイできます。このレシピでは、サイトを特定のプラットフォームにデプロイする方法はカバーしませんが、プロジェクト用にDockerをセットアップする方法を紹介します。
+
+## 前提条件
+
+- ローカルマシンにDockerがインストールされていること。[お使いのオペレーティング・システムのインストール手順はこちら](https://docs.docker.com/get-docker/)を参照してください。
+- プロジェクト内にDockerfileが存在すること。[Dockerfileについて詳しくはこちら](https://docs.docker.com/engine/reference/builder/)を参照し、以下のセクションのDockerfileを出発点として使うことができます。
+
+## Dockerfileを作成する
+
+プロジェクトのルートディレクトリに `Dockerfile` というファイルを作成します。このファイルにはあなたのニーズに応じて異なるサイトを構築するための指示が含まれています。このガイドではすべての可能なオプションを紹介することはできませんが、SSRと静的モードのスタートポイントを提供します。
```
Comment (morinokami) [inline src/content/docs/ja/recipes/docker.mdx:19]:
```suggestion
プロジェクトのルートディレクトリに`Dockerfile`というファイルを作成します。このファイルにはサイトを構築するための指示が含まれおり、ニーズに応じて内容は異なります。このガイドではすべての可能なオプションを紹介することはできませんが、SSRと静的モードのスタートポイントを提供します。
```

File: src/content/docs/ja/recipes/docker.mdx [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+
+npm以外のパッケージ・マネージャーを使っている場合は、それに応じてコマンドを調整する必要があります。
+
+### SSR
+
+このDockerfileはNode.jsを使用してサイトを構築し、ポート `4321` でサービスを提供します。そのため、Astroプロジェクトに[Nodeアダプタ](/ja/guides/integrations-guide/node/)がインストールされている必要があります。
```
Comment (morinokami) [inline src/content/docs/ja/recipes/docker.mdx:25]:
```suggestion
このDockerfileはNode.jsを使用してサイトを構築し、ポート`4321`でサービスを提供します。そのため、Astroプロジェクトに[Nodeアダプタ](/ja/guides/integrations-guide/node/)がインストールされている必要があります。
```

File: src/content/docs/ja/recipes/docker.mdx [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+
+```docker title="Dockerfile" del={1} add={2}
+FROM node:lts as runtime
+FROM node:lts-alpine as runtime
+```
+:::
+
+### .dockerignoreを追加する
+
+プロジェクトに `.dockerignore` ファイルを追加するのがベストプラクティスです。このファイルには、Dockerの `COPY` コマンドや `ADD` コマンドで無視するファイルやフォルダを記述します。これはビルドプロセスを高速化し、最終的なイメージのサイズを小さくします。
```
Comment (morinokami) [inline src/content/docs/ja/recipes/docker.mdx:53]:
```suggestion
プロジェクトに`.dockerignore`ファイルを追加するのがベストプラクティスです。このファイルには、Dockerの`COPY`コマンドや`ADD`コマンドで無視するファイルやフォルダを記述します。これはビルドプロセスを高速化し、最終的なイメージのサイズを小さくします。
```

File: src/content/docs/ja/recipes/docker.mdx [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+
+```docker title=".dockerignore"
+.DS_Store
+node_modules
+dist
+```
+
+このファイルは `Dockerfile` 自体と同じディレクトリに置く必要があります。[詳しくは `.dockerignore` ドキュメントを読んでください](https://docs.docker.com/engine/reference/builder/#dockerignore-file)
```
Comment (morinokami) [inline src/content/docs/ja/recipes/docker.mdx:61]:
```suggestion
このファイルは`Dockerfile`自体と同じディレクトリに置く必要があります。[詳しくは`.dockerignore`ドキュメントを読んでください](https://docs.docker.com/engine/reference/builder/#dockerignore-file)
```

File: src/content/docs/ja/recipes/docker.mdx [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+
+### Static
+
+#### Apache (httpd)
+
+以下のDockerfileはあなたのサイトを構築し、デフォルトの設定でポート `80` のApache htppdを使ってサービスを提供します。
```
Comment (morinokami) [inline src/content/docs/ja/recipes/docker.mdx:67]:
```suggestion
以下のDockerfileは、サイトをビルドし、それをApache htppdによりデフォルトの設定でポート`80`にて配信します。
```

File: src/content/docs/ja/recipes/docker.mdx [hunk 6]
```diff
@@ -0,0 +1,190 @@
...
+COPY . .
+RUN npm run build
+
+FROM nginx:alpine AS runtime
+COPY ./nginx/nginx.conf /etc/nginx/nginx.conf
+COPY --from=build /app/dist /usr/share/nginx/html
+EXPOSE 8080
+```
+
+上記のDockerfileをビルドするには、NGINX用の設定ファイルも作成する必要があります。プロジェクトのルートディレクトリに `nginx` というフォルダを作成し、その中に `nginx.conf` というファイルを作成します。
```
Comment (morinokami) [inline src/content/docs/ja/recipes/docker.mdx:101]:
```suggestion
上記のDockerfileをビルドするには、NGINX用の設定ファイルも作成する必要があります。プロジェクトのルートディレクトリに`nginx`というフォルダを作成し、その中に`nginx.conf`というファイルを作成します。
```

File: src/content/docs/ja/recipes/docker.mdx [hunk 7]
```diff
@@ -0,0 +1,190 @@
...
+    location / {
+            try_files $uri $uri/index.html =404;
+    }
+  }
+}
+```
+
+### マルチステージビルド（SSRを使用）
+
+Dockerの[マルチステージビルド](https://docs.docker.com/build/building/multi-stage/)のおかげで、ソースコードのみが変更された場合にnpmの依存関係を再インストールしないことで、あなたのサイトのビルドプロセスを最適化する、より高度なDockerfileの例です。これにより、依存関係のサイズにもよりますが、ビルド時間を数分でも短縮することができます。
```
Comment (morinokami) [inline src/content/docs/ja/recipes/docker.mdx:139]:
```suggestion
以下は、Dockerの[マルチステージビルド](https://docs.docker.com/build/building/multi-stage/)を用い、ソースコードのみが変更された場合にnpmの依存関係を再インストールしないようにすることで、サイトのビルドプロセスを最適化する、より高度なDockerfileの例です。これにより、依存関係のサイズにもよりますが、ビルド時間を場合によっては数分程度短縮できます。
```

File: src/content/docs/ja/recipes/docker.mdx [hunk 8]
```diff
@@ -0,0 +1,190 @@
...
+
+```docker title="Dockerfile"
+FROM node:lts AS base
+WORKDIR /app
+
+# package.jsonとpackage-lock.jsonのみをここにコピーすることで、以下の `-deps` ステップがソースコードから独立していることを保証する。
+# したがって、ソースコードが変更されただけであれば、 `-deps` ステップはスキップされる。
```
Comment (morinokami) [inline src/content/docs/ja/recipes/docker.mdx:146]:
```suggestion
# package.jsonとpackage-lock.jsonのみをここにコピーすることで、以下の`-deps`ステップがソースコードから独立していることが保証されます。
# したがって、ソースコードが変更されただけであれば、 `-deps`ステップはスキップされます。
```

File: src/content/docs/ja/recipes/docker.mdx [hunk 9]
```diff
@@ -0,0 +1,190 @@
...
+
+ENV HOST=0.0.0.0
+ENV PORT=4321
+EXPOSE 4321
+CMD node ./dist/server/entry.mjs
+```
+
+## レシピ
+
+1. プロジェクトのルート・ディレクトリで以下のコマンドを実行してコンテナをビルドします。`<your-astro-image-name>` には任意の名前を使用してください。
```
Comment (morinokami) [inline src/content/docs/ja/recipes/docker.mdx:171]:
```suggestion
1. プロジェクトのルートディレクトリで以下のコマンドを実行してコンテナをビルドします。`<your-astro-image-name>`には任意の名前を使用してください。
```

File: src/content/docs/ja/recipes/docker.mdx [hunk 10]
```diff
@@ -0,0 +1,190 @@
...
+
+```bash
+docker build -t <your-astro-image-name> .
+```
+
+これでイメージが出力され、ローカルで実行することも、好きなプラットフォームにデプロイすることもできます。
+
+2. イメージをローカル・コンテナとして実行するには、以下のコマンドを使用します。
```
Comment (morinokami) [inline src/content/docs/ja/recipes/docker.mdx:179]:
```suggestion
2. イメージをローカルコンテナとして実行するには、以下のコマンドを使用します。
```

File: src/content/docs/ja/recipes/docker.mdx [hunk 11]
```diff
@@ -0,0 +1,190 @@
...
+
+`<local-port>` を自分のマシンで開いているポートに置き換えてください。また `<container-port>`をDockerコンテナが公開しているポートに置き換えてください(上記の例では`4321`、`80`、`8080`)。
```
Comment (morinokami) [inline src/content/docs/ja/recipes/docker.mdx:181]:
```suggestion
`<local-port>`を自分のマシンの使用されていないポートに置き換えてください。また `<container-port>`をDockerコンテナが公開しているポートに置き換えてください（上記の例では`4321`、`80`、`8080`）。
```

File: src/content/docs/ja/recipes/docker.mdx [hunk 12]
```diff
@@ -0,0 +1,190 @@
...
+
+```bash
+docker run -p <local-port>:<container-port> <your-astro-image-name>
+```
+
+`http://localhost:<local-port>` でアクセスできるはずです。
```
Comment (morinokami) [inline src/content/docs/ja/recipes/docker.mdx:187]:
```suggestion
`http://localhost:<local-port>`からサイトにアクセスできるはずです。
```

File: src/content/docs/ja/recipes/docker.mdx [hunk 13]
```diff
@@ -0,0 +1,190 @@
...
+
+3. これでウェブサイトが正常にビルドされ、コンテナにパッケージ化されたので、クラウドプロバイダにデプロイすることができます。[Google Cloud](/ja/guides/deploy/google-cloud/#cloud-run-ssrと静的配信)のデプロイガイドを参照し、Dockerドキュメントの[Deploy your app](https://docs.docker.com/language/nodejs/deploy/)ページを参照してください。
```
Comment (morinokami) [inline src/content/docs/ja/recipes/docker.mdx:189]:
```suggestion
3. これでウェブサイトが正常にビルドされ、コンテナにパッケージ化されたので、クラウドプロバイダにデプロイすることができます。例として[Google Cloud](/ja/guides/deploy/google-cloud/#cloud-run-ssrと静的配信)のデプロイガイドや、Dockerドキュメントの[Deploy your app](https://docs.docker.com/language/nodejs/deploy/)ページを参照してください。
```

**Review by morinokami — CHANGES_REQUESTED** [review]
@sarah11918 Added some suggestions! I think we can merge this PR once those suggestions are merged.

@piro-hiroki Please let me know if you have anything to add!

**Review by sarah11918 — APPROVED** [review]
Thank you both @piro-hiroki and @morinokami ! I am so thankful for your contributions!

**Pull request merged**